src/directory_listing.rs

```diff
@@ -589,3 +588,70 @@ fn parse_last_modified(modified: SystemTime) -> Result<DateTime<Local>> {
         )),
     }
 }
+
+/// Formats the file size in bytes to a human-readable string
+fn format_file_size(size: u64) -> String {
+    let units = ["B", "KiB", "MiB", "GiB", "TiB", "PiB"];
+    let mut size_tmp = size;
+
+    for i in 0..units.len() - 1 {
+        if size_tmp < 1024 {
+            // return the size with the current unit
+            return format!("{} {}", size_tmp, units[i]);
+        } else if size_tmp < 1024 * 1024 && i < units.len() - 1 {
+            // return the size divided by 1024 with the next unit
+            return format!("{:.2} {}", size_tmp as f64 / 1024.0, units[i + 1]);
+        } 
+        size_tmp >>= 10;
+    }
+
+    // if size is too large, return the largest unit
+    format!("{} {}", size_tmp, units[units.len() - 1])
```

Comment:
We are going to have at most 4096 PiB in 64 bit, not large enough to strike the decimal points. So it’s probably best if we only iterate over `units[1 . .units.len() - 1]` – `units[0]` is handled as a special case up front and `units[units.len() - 1]` is a fallback unit to be used if the loop finishes without determining a suitable unit. You can really iterate over the values, not the indexes in that setup. And you’ll leave the loop with a value that you can divide by 1024.0 to display it.